### PR TITLE
Remove bootstrap.js and other unnecessary dependencies

### PR DESCRIPTION
**Pull request by aaxelb**
## Purpose
Remove unnecessary dependencies

## Summary of Changes
- Remove duplicate font-awesome, osf-style, bootstrap css
- Remove bootstrap js (from ember-cli-bootstrap-sassy)
  - Change anything that depended on it to use ember-bootstrap
  - Override ember-bootstrap's `bs-alert` component to use our preferred ✖️ 
- Change `ajax-helpers/authenticatedAjax` to return `RSVP.Promise` instead of a jQuery deferred, so it doesn't require special casing elsewhere.

## Side Effects / Testing Notes
Things that were touched that should still work just fine:
- navbar dropdowns
- navbar collapse on small screens
- alerts under the navbar (maintenance banner, status banners)
- institution carousel on the dashboard
- tooltips on "new and noteworthy" and "popular" project descriptions

## Ticket

N/A

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: app/utils/ajax-helpers.ts [hunk 1]
```diff
@@ -30,5 +31,5 @@ export default function authenticatedAJAX(options) {
             },
         });
     }
-    return $.ajax(options);
+    return new EmberPromise((resolve, reject) => $.ajax(options).then(resolve, reject));
```
Comment (binoculars) [inline app/utils/ajax-helpers.ts]: What does this do? Is wrapping it in an RSVP promise necessary?

Comment (jamescdavis) [reply app/utils/ajax-helpers.ts]: See: https://github.com/CenterForOpenScience/ember-osf-web/pull/179/files#diff-61cf47780bd3aa2cfdc9e00027572539L74

This handles that generally.

File: app/components/osf-navbar/auth-dropdown/component.ts [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
  * Display the login dropdown on the navbar
  *
  * @class osf-navbar/auth-dropdown
  */
-export default class NavbarAuthDropdown extends Component.extend({
-    tagName: 'li',
-    classNames: ['dropdown', 'secondary-nav-dropdown'],
-    classNameBindings: ['notAuthenticated:sign-in'],
-}) {
+@tagName('')
```
[the remark this answers is not in the document]
Comment (aaxelb) [reply app/components/osf-navbar/auth-dropdown/component.ts]: Yeah that's deliberate, it's a tagless component. osf-style makes specific assumptions about navbar structure (such great shared styles), so the `li` can't contain the dropdown; it has to *be* the dropdown, and be a direct child of `.navbar-nav`. [Switching to `bs-dropdown`](https://github.com/CenterForOpenScience/ember-osf-web/pull/179/files#diff-e4f529ea7d973a2dba03e5f092096ac9R2), along with the `li` [sometimes not being a dropdown](https://github.com/CenterForOpenScience/ember-osf-web/pull/179/files#diff-e4f529ea7d973a2dba03e5f092096ac9R43), meant it made more sense to me to make this component tagless.

File: app/utils/ajax-helpers.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import config from 'ember-get-config';
 import $ from 'jquery';
+import { Promise as EmberPromise } from 'rsvp';
```
Comment (jamescdavis) [inline app/utils/ajax-helpers.ts]: Maybe just `import RSVP from 'rsvp''` here and `return new RSVP.Promise(...` below?

Comment (aaxelb) [reply app/utils/ajax-helpers.ts]: Yeah, that's cleaner 👍

File: app/models/osf-model.ts [hunk 4]
```diff
@@ -3,7 +3,6 @@ import { get } from '@ember/object';
 import { alias } from '@ember/object/computed';
 import PromiseProxyMixin from '@ember/object/promise-proxy-mixin';
 import { merge } from '@ember/polyfills';
-import { bind } from '@ember/runloop';
 import DS from 'ember-data';
 import authenticatedAJAX from 'ember-osf-web/utils/ajax-helpers';
 import { Promise as EmberPromise } from 'rsvp';
```
Comment (jamescdavis) [inline app/models/osf-model.ts]: also here

Comment (jamescdavis) [reply app/models/osf-model.ts]: This will be refactored to not use a promise in another PR.

File: ember-cli-build.js [hunk 5]
```diff
@@ -71,7 +71,6 @@ module.exports = function(defaults) {
 
     app.import('node_modules/dropzone/dist/dropzone.css');
     app.import('node_modules/dropzone/dist/dropzone.js');
-    app.import('node_modules/osf-style/css/base.css');
     app.import('node_modules/@centerforopenscience/osf-style/css/base.css');
```
Comment (jamescdavis) [inline ember-cli-build.js]: I don't think we need this since we are already importing the scss from osf-style.

Comment (jamescdavis) [reply ember-cli-build.js]: Just removed and rebuilt to make sure. Confirmed we don't need it (and it's redundant).

**Pull request merged**